### PR TITLE
Remove SwapDirection

### PR DESCRIPTION
**Pull request by lubkoll**

### CODE REVIEW

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs [hunk 1]
```diff
@@ -105,40 +99,30 @@ pub fn execute_swap_non_vault_funds(
 pub fn calculate_swap_amount(
     contract_address: Addr,
     pool_config: PoolConfig,
-    swap_direction: SwapDirection,
-    token_in_amount: Uint128,
+    offer: Coin,
```
Comment (magiodev) [inline smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs:102]: Is a bit confusing the way we call tokens, which sometimes is token_in and here is offer.

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs [hunk 2]
```diff
@@ -262,61 +262,77 @@ fn do_swap_deposit_merge(
...
     };
 
     let dex_router = DEX_ROUTER.may_load(deps.storage)?;
-    let twap_price = get_twap_price(deps.storage, &deps.querier, &env, twap_window_seconds)?;
     let swap_calc_result = calculate_swap_amount(
         env.contract.address,
         pool_config,
-        swap_direction,
-        token_in_amount,
+        token_in,
```
Comment (magiodev) [inline smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs:331]: Here is token_in

**Review by magiodev — APPROVED** [review]
Mostly looks good, just a minor comment.

File: smart-contracts/osmosis/contracts/cl-vault/src/contract.rs [hunk 3]
```diff
@@ -292,8 +290,6 @@ pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, Co
     OLD_POSITION.remove(deps.storage);
     #[allow(deprecated)]
     CURRENT_BALANCE.remove(deps.storage);
-    #[allow(deprecated)]
-    CURRENT_SWAP.remove(deps.storage);
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/contract.rs:296]: Afaik this remove was not migrated yet to mainnet contracts, if so we should inline the remove so we don't orphan state items

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs [hunk 4]
```diff
@@ -78,57 +75,70 @@ pub(crate) fn execute_any_deposit(
...
     }
 
-    // Swap logic
-    // TODO_FUTURE: Optimize this if conditions
-    let (swap_amount, swap_direction, left_over_amount) = if !deposit_info.base_refund.is_zero() {
-        // range is above current tick
-        let swap_amount = if pool_details.current_tick > position.upper_tick {
-            deposit_info.base_refund
+    let twap_price = get_twap_price(deps.storage, &deps.querier, &env, 24u64)?;
+    let (offer, out_denom, remainder, price) = if !deposit_info.base_refund.is_zero() {
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs:79]: If we change offer according to Stefano's comments, lets change it here to so we have token_in, out_denom

**Review by LaurensKubat — COMMENTED** [review]
generally LGTM, 2 small comments

File: smart-contracts/osmosis/contracts/cl-vault/src/contract.rs (hunk 3 above)
Comment (lubkoll) [reply smart-contracts/osmosis/contracts/cl-vault/src/contract.rs:296]: I don't get what you refer to with "orphaning" items, what is that and what is the real problem with removing this?

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs (hunk 4 above)
Comment (lubkoll) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs]: done

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs (hunk 2 above)
Comment (lubkoll) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs:331]: done

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs (hunk 1 above)
Comment (lubkoll) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs]: done

**Comment by lubkoll** [comment]
I removed some failing code in a test-tube test (in authz::exact_deposit_withdraw_equal).
Happy to add it back if somebody is able to explain what is tested there and why we need this test.

**Comment by lubkoll** [comment]
> I removed some failing code in a test-tube test (in authz::exact_deposit_withdraw_equal). Happy to add it back if somebody is able to explain what is tested there and why we need this test.

Added the test back..

**Review by ajansari95 — APPROVED** [review]
lgtm

**Pull request merged**